README.md

```diff
@@ -42,6 +42,9 @@ specify the `prefix` environment variable.  For example, to install to
 
 Below are some system-specific build instructions for this repository; pull
 requests with steps for other environments are certainly welcome.
+([Erhannis:] The easiest way I've found to build this, as of 2022, is to
+install Ubuntu 14.04.6 in VirtualBox and follow the Debian 8 instructions.
```

Comment:
have you tried using the dockerfile in this repository?  that might be a little more convenient, as it lets you compile code on whatever system you prefer to develop on without having to install a bunch of older tools.  I haven't tried it in a while but IIRC the dockerfile in this repository worked pretty well.

I'd prefer not to promote Oracle's VM product, I'm kind of holding a grudge.

Reply:
Oh - I didn't realize there was a docker, and/or don't really know how to use docker, and the README didn't mention it.  I cd'd into ./docker/debian, and did `docker build .`, which succeeded - but I don't know how to use it.  I tried checking out the Apache Thrift tag you give below, thinking maybe it'd explain, but it wasn't clear there how to use the dockerfile, either.  I saw ./contrib/nestedvm/docker/build.sh, so I cd'd into that directory and ran it, but the directory references seemed wrong, so I cd'd one folder up into ./contrib/nestedvm and ran `./docker/build.sh`, and it got farther than the first time, but then for
```
RUN apt-get -y install     ant ant-optional maven flex bison pkg-config libtool automake wget     --no-install-recommends
```
it gave me a bunch of 404s; and that's where I am now.  Logs below:
```
Step 3/7 : RUN apt-get -y install     ant ant-optional maven flex bison pkg-config libtool automake wget     --no-install-recommends
 ---> Running in e6f86f12034a
Reading package lists...
Building dependency tree...
Reading state information...
The following extra packages will be installed:
  autoconf autotools-dev bsh file java-wrappers junit junit4 libaether-java
  libaopalliance-java libapache-pom-java libasm3-java libasm4-java
  libasync-http-client-java libatinject-jsr330-api-java
  libavalon-framework-java libbatik-java libbison-dev libbsf-java libbsh-java
  libcdi-api-java libcglib3-java libclassworlds-java libcommons-cli-java
  libcommons-codec-java libcommons-configuration-java
  libcommons-httpclient-java libcommons-io-java libcommons-lang-java
  libcommons-logging-java libcommons-net2-java libcommons-parent-java
  libdoxia-java libeasymock-java libfl-dev libfop-java libganymed-ssh2-java
  libgeronimo-interceptor-3.0-spec-java libguava-java libhamcrest-java
  libhttpclient-java libhttpcore-java libicu52 libitext1-java
  libjackrabbit-java libjaxen-java libjaxp1.3-java libjdom1-java libjetty-java
  libjline-java libjsch-agent-proxy-java libjsch-java libjsoup-java
  libjsr305-java libjzlib-java liblog4j1.2-java libmagic1 libmaven-parent-java
  libmaven-scm-java libmaven2-core-java libmodello-java
  libnetbeans-cvsclient-java libnetty-java libobjenesis-java
  libplexus-ant-factory-java libplexus-archiver-java
  libplexus-bsh-factory-java libplexus-build-api-java libplexus-cipher-java
  libplexus-classworlds-java libplexus-classworlds2-java libplexus-cli-java
  libplexus-container-default-java libplexus-containers-java
  libplexus-containers1.5-java libplexus-i18n-java
  libplexus-interactivity-api-java libplexus-interpolation-java
  libplexus-io-java libplexus-sec-dispatcher-java libplexus-utils-java
  libplexus-utils2-java libpsl0 libqdox-java libregexp-java libservlet2.5-java
  libservlet3.1-java libsigsegv2 libsisu-guice-java libsisu-ioc-java
  libslf4j-java libwagon-java libwagon2-java libxalan2-java libxbean-java
  libxerces2-java libxml-commons-external-java libxml-commons-resolver1.1-java
  libxmlgraphics-commons-java libxt6 m4 unzip
Suggested packages:
  ant-doc ant-gcj ant-optional-gcj antlr javacc jython libbcel-java
  libcommons-net-java libmail-java libjdepend-java liboro-java
  autoconf-archive gnu-standards autoconf-doc gettext bison-doc bsh-doc
  libservlet3.0-java junit-doc libaopalliance-java-doc
  libasync-http-client-java-doc libatinject-jsr330-api-java-doc
  libavalon-framework-java-doc libclassworlds-java-doc
  libcommons-httpclient-java-doc libcommons-io-java-doc
  libcommons-lang-java-doc libcommons-logging-java-doc
  libexcalibur-logkit-java libcommons-net2-java-doc libdoxia-java-doc
  libeasymock-java-doc libjaxp1.3-java-gcj libjdom1-java-doc jetty
  libjetty-java-doc libjline-java-doc libjsoup-java-doc libjsr305-java-doc
  liblog4j1.2-java-doc libmaven-scm-java-doc libmodello-java-doc maven2
  libobjenesis-java-doc libplexus-build-api-java-doc libplexus-cipher-java-doc
  libplexus-classworlds-java-doc libplexus-classworlds2-java-doc
  libplexus-cli-java-doc libplexus-container-default-java-doc
  libplexus-i18n-java-doc libplexus-interactivity-api-java-doc
  libplexus-interpolation-java-doc libplexus-sec-dispatcher-java-doc
  libplexus-utils-java-doc libplexus-utils2-java-doc libqdox-java-doc
  libtool-doc automaken gfortran fortran95-compiler gcj-jdk libwagon-java-doc
  libxalan2-java-doc libxsltc-java groovy libspring-core-java
  libspring-beans-java libspring-context-java libspring-web-java
  libequinox-osgi-java libxerces2-java-doc libxerces2-java-gcj
  libxml-commons-resolver1.1-java-doc libxmlgraphics-commons-java-doc zip
Recommended packages:
  fop rhino default-jre libcommons-beanutils-java libcommons-collections3-java
  libcommons-digester-java libcommons-jexl2-java libcommons-jxpath-java
  libcommons-vfs-java libsaxon-java icc-profiles-free libdom4j-java
  libxom-java libosgi-compendium-java libosgi-core-java testng libltdl-dev
The following NEW packages will be installed:
  ant ant-optional autoconf automake autotools-dev bison bsh file flex
  java-wrappers junit junit4 libaether-java libaopalliance-java
  libapache-pom-java libasm3-java libasm4-java libasync-http-client-java
  libatinject-jsr330-api-java libavalon-framework-java libbatik-java
  libbison-dev libbsf-java libbsh-java libcdi-api-java libcglib3-java
  libclassworlds-java libcommons-cli-java libcommons-codec-java
  libcommons-configuration-java libcommons-httpclient-java libcommons-io-java
  libcommons-lang-java libcommons-logging-java libcommons-net2-java
  libcommons-parent-java libdoxia-java libeasymock-java libfl-dev libfop-java
  libganymed-ssh2-java libgeronimo-interceptor-3.0-spec-java libguava-java
  libhamcrest-java libhttpclient-java libhttpcore-java libicu52 libitext1-java
  libjackrabbit-java libjaxen-java libjaxp1.3-java libjdom1-java libjetty-java
  libjline-java libjsch-agent-proxy-java libjsch-java libjsoup-java
  libjsr305-java libjzlib-java liblog4j1.2-java libmagic1 libmaven-parent-java
  libmaven-scm-java libmaven2-core-java libmodello-java
  libnetbeans-cvsclient-java libnetty-java libobjenesis-java
  libplexus-ant-factory-java libplexus-archiver-java
  libplexus-bsh-factory-java libplexus-build-api-java libplexus-cipher-java
  libplexus-classworlds-java libplexus-classworlds2-java libplexus-cli-java
  libplexus-container-default-java libplexus-containers-java
  libplexus-containers1.5-java libplexus-i18n-java
  libplexus-interactivity-api-java libplexus-interpolation-java
  libplexus-io-java libplexus-sec-dispatcher-java libplexus-utils-java
  libplexus-utils2-java libpsl0 libqdox-java libregexp-java libservlet2.5-java
  libservlet3.1-java libsigsegv2 libsisu-guice-java libsisu-ioc-java
  libslf4j-java libtool libwagon-java libwagon2-java libxalan2-java
  libxbean-java libxerces2-java libxml-commons-external-java
  libxml-commons-resolver1.1-java libxmlgraphics-commons-java libxt6 m4 maven
  pkg-config unzip wget
0 upgraded, 110 newly installed, 0 to remove and 1 not upgraded.
Need to get 49.5 MB of archives.
After this operation, 109 MB of additional disk space will be used.
Err http://deb.debian.org/debian/ jessie/main libmagic1 amd64 1:5.22+15-2+deb8u3
  404  Not Found
Get:1 http://deb.debian.org/debian/ jessie/main libsigsegv2 amd64 2.10-4+b1 [29.2 kB]
Get:2 http://deb.debian.org/debian/ jessie/main m4 amd64 1.4.17-4 [254 kB]
Get:3 http://deb.debian.org/debian/ jessie/main libfl-dev amd64 2.5.39-8+deb8u2 [75.3 kB]
Get:4 http://deb.debian.org/debian/ jessie/main flex amd64 2.5.39-8+deb8u2 [421 kB]
Err http://deb.debian.org/debian/ jessie/main libicu52 amd64 52.1-8+deb8u6
  404  Not Found
Get:5 http://deb.debian.org/debian/ jessie/main libxt6 amd64 1:1.1.4-1+b1 [186 kB]
Get:6 http://deb.debian.org/debian/ jessie/main libpsl0 amd64 0.5.1-1 [41.6 kB]
Err http://deb.debian.org/debian/ jessie/main wget amd64 1.16-1+deb8u4
  404  Not Found
Err http://security.debian.org/ jessie/updates/main wget amd64 1.16-1+deb8u4
  404  Not Found [IP: 151.101.130.132 80]
Err http://deb.debian.org/debian/ jessie/main file amd64 1:5.22+15-2+deb8u3
  404  Not Found
Get:7 http://deb.debian.org/debian/ jessie/main ant all 1.9.4-3 [1888 kB]
Get:8 http://deb.debian.org/debian/ jessie/main ant-optional all 1.9.4-3 [316 kB]
Get:9 http://deb.debian.org/debian/ jessie/main autoconf all 2.69-8 [340 kB]
Get:10 http://deb.debian.org/debian/ jessie/main autotools-dev all 20140911.1 [70.5 kB]
Get:11 http://deb.debian.org/debian/ jessie/main automake all 1:1.14.1-4+deb8u1 [724 kB]
Get:12 http://deb.debian.org/debian/ jessie/main libbison-dev amd64 2:3.0.2.dfsg-2 [426 kB]
Get:13 http://deb.debian.org/debian/ jessie/main bison amd64 2:3.0.2.dfsg-2 [765 kB]
Get:14 http://deb.debian.org/debian/ jessie/main libjline-java all 1.0-2 [69.5 kB]
Get:15 http://deb.debian.org/debian/ jessie/main libbsh-java all 2.0b4-15+deb8u1 [266 kB]
Get:16 http://deb.debian.org/debian/ jessie/main bsh all 2.0b4-15+deb8u1 [8094 B]
Get:17 http://deb.debian.org/debian/ jessie/main unzip amd64 6.0-16+deb8u3 [162 kB]
Get:18 http://deb.debian.org/debian/ jessie/main java-wrappers all 0.1.28 [10.7 kB]
Get:19 http://deb.debian.org/debian/ jessie/main junit all 3.8.2-8 [107 kB]
Get:20 http://deb.debian.org/debian/ jessie/main libhamcrest-java all 1.3-5 [387 kB]
Get:21 http://deb.debian.org/debian/ jessie/main junit4 all 4.11-3 [224 kB]
Get:22 http://deb.debian.org/debian/ jessie/main libnetty-java all 1:3.2.6.Final-2 [671 kB]
Get:23 http://deb.debian.org/debian/ jessie/main libslf4j-java all 1.7.7-1 [112 kB]
Get:24 http://deb.debian.org/debian/ jessie/main libasync-http-client-java all 1.6.5-3 [307 kB]
Get:25 http://deb.debian.org/debian/ jessie/main libplexus-classworlds2-java all 2.5.2-1 [46.7 kB]
Get:26 http://deb.debian.org/debian/ jessie/main libasm3-java all 3.3.2-2 [670 kB]
Get:27 http://deb.debian.org/debian/ jessie/main libapache-pom-java all 10-2 [3870 B]
Get:28 http://deb.debian.org/debian/ jessie/main libcommons-parent-java all 22-2 [6786 B]
Get:29 http://deb.debian.org/debian/ jessie/main libcommons-lang-java all 2.6-4 [269 kB]
Get:30 http://deb.debian.org/debian/ jessie/main libcommons-cli-java all 1.2-3 [39.2 kB]
Get:31 http://deb.debian.org/debian/ jessie/main libatinject-jsr330-api-java all 1.0+ds1-1 [22.6 kB]
Get:32 http://deb.debian.org/debian/ jessie/main libjsr305-java all 0.1~+svn49-4 [28.1 kB]
Get:33 http://deb.debian.org/debian/ jessie/main libguava-java all 17.0-1 [1965 kB]
Get:34 http://deb.debian.org/debian/ jessie/main libjaxen-java all 1.1.6-1 [209 kB]
Get:35 http://deb.debian.org/debian/ jessie/main libjdom1-java all 1.1.3-1 [150 kB]
Get:36 http://deb.debian.org/debian/ jessie/main libclassworlds-java all 1.1-final-5 [39.3 kB]
Get:37 http://deb.debian.org/debian/ jessie/main libcommons-logging-java all 1.2-1 [62.1 kB]
Get:38 http://deb.debian.org/debian/ jessie/main libcommons-configuration-java all 1.10-2 [331 kB]
Get:39 http://deb.debian.org/debian/ jessie/main libxml-commons-resolver1.1-java all 1.2-7 [91.9 kB]
Get:40 http://deb.debian.org/debian/ jessie/main libxml-commons-external-java all 1.4.01-2 [245 kB]
Get:41 http://deb.debian.org/debian/ jessie/main libxerces2-java all 2.11.0-7 [1365 kB]
Get:42 http://deb.debian.org/debian/ jessie/main libjaxp1.3-java all 1.3.05-2 [450 kB]
Get:43 http://deb.debian.org/debian/ jessie/main libxalan2-java all 2.7.1-9 [3169 kB]
Get:44 http://deb.debian.org/debian/ jessie/main libbsf-java all 1:2.4.0-5 [72.2 kB]
Get:45 http://deb.debian.org/debian/ jessie/main libavalon-framework-java all 4.2.0-9 [70.6 kB]
Get:46 http://deb.debian.org/debian/ jessie/main libcommons-io-java all 2.4-2 [176 kB]
Err http://deb.debian.org/debian/ jessie/main libbatik-java all 1.7+dfsg-5
  404  Not Found
Get:47 http://deb.debian.org/debian/ jessie/main libxmlgraphics-commons-java all 1.5-4 [549 kB]
Get:48 http://deb.debian.org/debian/ jessie/main libfop-java all 1:1.1.dfsg2-1+deb8u1 [3199 kB]
Get:49 http://deb.debian.org/debian/ jessie/main libcommons-codec-java all 1.9-1 [215 kB]
Get:50 http://deb.debian.org/debian/ jessie/main libhttpcore-java all 4.3.3-1 [539 kB]
Get:51 http://deb.debian.org/debian/ jessie/main libhttpclient-java all 4.3.5-2 [549 kB]
Get:52 http://deb.debian.org/debian/ jessie/main libitext1-java all 1.4-6 [1194 kB]
Get:53 http://deb.debian.org/debian/ jessie/main liblog4j1.2-java all 1.2.17-5 [386 kB]
Get:54 http://deb.debian.org/debian/ jessie/main libplexus-interpolation-java all 1.19-1 [60.4 kB]
Err http://deb.debian.org/debian/ jessie/main libplexus-utils-java all 1:1.5.15-4
  404  Not Found
Get:55 http://deb.debian.org/debian/ jessie/main libplexus-container-default-java all 1.0-alpha-9-stable-1-7 [150 kB]
Get:56 http://deb.debian.org/debian/ jessie/main libplexus-classworlds-java all 1.5.0-5 [48.2 kB]
Get:57 http://deb.debian.org/debian/ jessie/main libxbean-java all 4.0-2 [815 kB]
Get:58 http://deb.debian.org/debian/ jessie/main libplexus-containers-java all 1.0~beta3.0.7-7 [183 kB]
Get:59 http://deb.debian.org/debian/ jessie/main libplexus-i18n-java all 1.0-beta-10-3 [13.5 kB]
Get:60 http://deb.debian.org/debian/ jessie/main libdoxia-java all 1.1.4-2 [731 kB]
Get:61 http://deb.debian.org/debian/ jessie/main libplexus-build-api-java all 0.0.7-1 [9810 B]
Get:62 http://deb.debian.org/debian/ jessie/main libmodello-java all 1.1-3 [239 kB]
Get:63 http://deb.debian.org/debian/ jessie/main libplexus-ant-factory-java all 1.0~alpha2.1-3 [11.2 kB]
Get:64 http://deb.debian.org/debian/ jessie/main libplexus-io-java all 1.0~alpha5-2 [44.7 kB]
Err http://deb.debian.org/debian/ jessie/main libplexus-archiver-java all 1.2-1
  404  Not Found
Get:65 http://deb.debian.org/debian/ jessie/main libplexus-bsh-factory-java all 1.0~alpha7-3 [7874 B]
Get:66 http://deb.debian.org/debian/ jessie/main libplexus-interactivity-api-java all 1.0-alpha-6-7 [14.1 kB]
Get:67 http://deb.debian.org/debian/ jessie/main libplexus-cipher-java all 1.5-4 [14.8 kB]
Get:68 http://deb.debian.org/debian/ jessie/main libplexus-sec-dispatcher-java all 1.3.1-6 [27.0 kB]
Get:69 http://deb.debian.org/debian/ jessie/main libcommons-httpclient-java all 3.1-11+deb8u1 [298 kB]
Get:70 http://deb.debian.org/debian/ jessie/main libcommons-net2-java all 2.2-2 [193 kB]
Get:71 http://deb.debian.org/debian/ jessie/main libasm4-java all 5.0.3-1 [251 kB]
Get:72 http://deb.debian.org/debian/ jessie/main libcglib3-java all 3.1+dfsg-1 [726 kB]
Get:73 http://deb.debian.org/debian/ jessie/main libobjenesis-java all 1.2+full-3 [30.3 kB]
Get:74 http://deb.debian.org/debian/ jessie/main libeasymock-java all 3.2+ds-4 [84.8 kB]
Get:75 http://deb.debian.org/debian/ jessie/main libservlet2.5-java all 6.0.45+dfsg-1~deb8u1 [220 kB]
Get:76 http://deb.debian.org/debian/ jessie/main libjetty-java all 6.1.26-4 [781 kB]
Get:77 http://deb.debian.org/debian/ jessie/main libjzlib-java all 1.1.3-1 [69.9 kB]
Get:78 http://deb.debian.org/debian/ jessie/main libjsch-java all 0.1.51-1 [268 kB]
Get:79 http://deb.debian.org/debian/ jessie/main libjsoup-java all 1.8.1-1 [292 kB]
Get:80 http://deb.debian.org/debian/ jessie/main libganymed-ssh2-java all 250-2 [389 kB]
Get:81 http://deb.debian.org/debian/ jessie/main libnetbeans-cvsclient-java all 6.5-2 [349 kB]
Get:82 http://deb.debian.org/debian/ jessie/main libregexp-java all 1.5-3 [35.5 kB]
Get:83 http://deb.debian.org/debian/ jessie/main libmaven-scm-java all 1.3-5 [894 kB]
Get:84 http://deb.debian.org/debian/ jessie/main libwagon-java all 1.0.0-5 [203 kB]
Get:85 http://deb.debian.org/debian/ jessie/main libmaven2-core-java all 2.2.1-17 [697 kB]
Get:86 http://deb.debian.org/debian/ jessie/main libplexus-cli-java all 1.2-5 [9348 B]
Err http://deb.debian.org/debian/ jessie/main libplexus-utils2-java all 3.0.15-1
  404  Not Found
Get:87 http://deb.debian.org/debian/ jessie/main libqdox-java all 1.12-1 [168 kB]
Get:88 http://deb.debian.org/debian/ jessie/main libplexus-containers1.5-java all 1.5.5-6 [285 kB]
Get:89 http://deb.debian.org/debian/ jessie/main libaopalliance-java all 20070526-5 [9192 B]
Get:90 http://deb.debian.org/debian/ jessie/main libgeronimo-interceptor-3.0-spec-java all 1.0.1-1 [5028 B]
Get:91 http://deb.debian.org/debian/ jessie/main libservlet3.1-java all 8.0.14-1+deb8u11 [393 kB]
Get:92 http://deb.debian.org/debian/ jessie/main libcdi-api-java all 1.0-2 [31.9 kB]
Get:93 http://deb.debian.org/debian/ jessie/main libsisu-guice-java all 3.2.4-1 [614 kB]
Get:94 http://deb.debian.org/debian/ jessie/main libsisu-ioc-java all 2.3.0-5 [458 kB]
Get:95 http://deb.debian.org/debian/ jessie/main libaether-java all 1.13.1-2 [495 kB]
Get:96 http://deb.debian.org/debian/ jessie/main libjackrabbit-java all 2.3.6-1+deb8u2 [276 kB]
Get:97 http://deb.debian.org/debian/ jessie/main libjsch-agent-proxy-java all 0.0.7-1 [23.1 kB]
Get:98 http://deb.debian.org/debian/ jessie/main libmaven-parent-java all 21-2 [5744 B]
Get:99 http://deb.debian.org/debian/ jessie/main libtool all 2.4.2-1.11 [190 kB]
Get:100 http://deb.debian.org/debian/ jessie/main libwagon2-java all 2.7-1 [2021 kB]
Get:101 http://deb.debian.org/debian/ jessie/main maven all 3.0.5-3 [1281 kB]
Get:102 http://deb.debian.org/debian/ jessie/main pkg-config amd64 0.28-1 [57.5 kB]
E: Failed to fetch http://deb.debian.org/debian/pool/main/f/file/libmagic1_5.22+15-2+deb8u3_amd64.deb  404  Not Found

E: Failed to fetch http://deb.debian.org/debian/pool/main/i/icu/libicu52_52.1-8+deb8u6_amd64.deb  404  Not Found

E: Failed to fetch http://security.debian.org/pool/updates/main/w/wget/wget_1.16-1+deb8u4_amd64.deb  404  Not Found [IP: 151.101.130.132 80]

E: Failed to fetch http://deb.debian.org/debian/pool/main/f/file/file_5.22+15-2+deb8u3_amd64.deb  404  Not Found

E: Failed to fetch http://deb.debian.org/debian/pool/main/b/batik/libbatik-java_1.7+dfsg-5_all.deb  404  Not Found

E: Failed to fetch http://deb.debian.org/debian/pool/main/p/plexus-utils/libplexus-utils-java_1.5.15-4_all.deb  404  Not Found

E: Failed to fetch http://deb.debian.org/debian/pool/main/p/plexus-archiver/libplexus-archiver-java_1.2-1_all.deb  404  Not Found

E: Failed to fetch http://deb.debian.org/debian/pool/main/p/plexus-utils2/libplexus-utils2-java_3.0.15-1_all.deb  404  Not Found

E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
Fetched 38.5 MB in 19s (2017 kB/s)
The command '/bin/sh -c apt-get -y install     ant ant-optional maven flex bison pkg-config libtool automake wget     --no-install-recommends' returned a non-zero code: 100
Unable to find image 'thrift-nestedvm:latest' locally
docker: Error response from daemon: pull access denied for thrift-nestedvm, repository does not exist or may require 'docker login': denied: requested access to the resource is denied.
See 'docker run --help'.
```